lib/Sema/CSSimplify.cpp

```diff
@@ -1352,16 +1352,33 @@ ConstraintSystem::matchTupleTypes(TupleType *tuple1, TupleType *tuple2,
 // corresponding function type representations and the given match kind.
 static bool matchFunctionRepresentations(FunctionTypeRepresentation rep1,
                                          FunctionTypeRepresentation rep2,
-                                         ConstraintKind kind) {
+                                         ConstraintKind kind,
+                                         ConstraintLocator *locator) {
   switch (kind) {
   case ConstraintKind::Bind:
   case ConstraintKind::BindParam:
   case ConstraintKind::BindToPointerType:
   case ConstraintKind::Equal:
     return rep1 != rep2;
+      
+  case ConstraintKind::Subtype: {
+    if (!locator->isLastElement<LocatorPathElt::FunctionArgument>()) 
+      return false;
+    
+    auto isThick = [](FunctionTypeRepresentation rep) {
+      return rep == FunctionTypeRepresentation::Swift ||
+          rep == swift::FunctionTypeRepresentation::Block;
+    };
+
+    // Allowing all to "thick" (swift, block) conventions
+    // "thin" (c, thin) to "thick" or "thick" to "thick"
+    if (isThick(rep2))
```

Comment:
Since this just a single or condition, I think it makes sense to just inline it into the check instead of using a closure.